src/pages/api/auth/[...nextauth].ts

```diff
@@ -15,13 +18,20 @@ const sendVerificationRequest = async (
   params: SendVerificationRequestParams
 ) => {
   const { identifier, url } = params;
-  const { host } = new URL(url);
+  const { searchParams } = new URL(url);
+  // the current url
+  const callbackUrl = searchParams.get('callbackUrl');
+
   await emailClient.sendMail({
     to: identifier,
     from: OUR_EMAIL_FROM_FORMATTED,
     subject: 'Vegan Hacktivists Playground login',
-    text: verificationMail(url, true),
-    html: verificationMail(url),
+    text: callbackUrl?.includes('signin')
+      ? verificationMail(url, true)
+      : verifyRequestEmail(url, true),
+    html: callbackUrl?.includes('signin')
+      ? verificationMail(url)
+      : verifyRequestEmail(url),
   });
```

Comment:
I'd extract the condition, especially since it's used twice e.g.

`
  const getMailBody = callbackUrl?.includes('signin')
    ? verificationMail
    : verifyRequestEmail;
`



`
    { text: getMailBody(url, true),
    html: getMailBody(url) }
`

Reply:
Makes sense, implemented.